uptane-whitepaper.md

```diff
@@ -1 +1,106 @@
+- *Explicit and implicit revocation of keys:* providing a mechanism for replacing compromised keys so malevolent parties can not continue signing metadata to authenticate malware, and ensuring that keys are not used forever.
+- *Keeping the most vulnerable keys offline:* mandating that certain signing keys be unavailable to online services, thus making them harder to steal or compromise.
+
+These principles are extracted from an established security protocol called [The Update Framework](https://theupdateframework.io/). But very early on in Uptane's development, researchers realized some changes would be needed to adapt these principles for automotive ECUs. The first is the addition of a second repository, which divides labor and responsibility for different aspects of the verification process. The image repository holds an accurate inventory of all the images currently on all ECUs on all vehicles maintained by an OEM, and the corresponding metadata. This repository uses offline keys to sign all metadata, making it much harder for attackers to substitute compromised images. The director repository, which instructs vehicles as to what updates should be installed next, uses online keys to sign its metadata, allowing for easier and faster turnarounds. By combining the two repositories, an OEM can provide both customizability and security for the control units on their vehicles.
+
+Another modification made to the basic TUF design has to do with the way Uptane verifies the images, meaning the media used to supply the code and data that control the actions of the ECUs. In the verification step, the ECU determines if a file is safe to download by checking its accompanying metadata. As mentioned above, an ECU can use two different verification strategies, depending on its power. Verification can be full, which requires checking that the hashes and sizes of images in the signed metadata match the hashes and sizes stored on the image repository, or partial, which requires only the signature on the targets metadata file from the director repository to be checked.
+
+## **Basic Uptane Design** 
+This diagram illustrates how the checks and balances of this system works. The connected components on the right hand side of the diagram are on the vehicle, while the components on the left hand-side represent the repositories. The Image Repository can be thought of as an unchanging source of information about images. It is the keeper of every image currently deployed by the OEM, along with the metadata files that prove their authenticity. The Director Repository knows what software should be distributed to each ECU, given the current state of the repository. Since many ECUs do not have clocks, the Time Server informs vehicles about the current time in a cryptographically secure way, by receiving a list of tokens from vehicles, and returning a signed sequence that includes the token and the current time.
```

Comment:
```suggestion
This diagram illustrates how the checks and balances of this system works. The connected components on the right hand side of the diagram are on the vehicle, while the components on the left hand-side represent the repositories. The Image repository can be thought of as an unchanging source of information about images. It is the keeper of every image currently deployed by the OEM, along with the metadata files that prove their authenticity. The Director repository knows what software should be distributed to each ECU, given the current state of the repository. Since many ECUs do not have clocks, the Time Server informs vehicles about the current time in a cryptographically secure way, by receiving a list of tokens from vehicles, and returning a signed sequence that includes the token and the current time.
```
Do we really need to go into the Time Server here? That's a deployment issue that is just one way of managing secure time.

Reply:
Agreed. I was looking at that this morning as well. I'll take it out.
```suggestion
This diagram illustrates how the checks and balances of this system works. The connected components on the right hand side of the diagram are on the vehicle, while the components on the left hand-side represent the repositories. The Image Repository can be thought of as an unchanging source of information about images. It is the keeper of every image currently deployed by the OEM, along with the metadata files that prove their authenticity. The Director Repository knows what software should be distributed to each ECU, given the current state of the repository. 
```